projects/storefrontlib/cms-components/myaccount/my-coupons/claim-dialog/claim-dialog.component.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+  ElementRef,
+  HostListener,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { Subscription } from 'rxjs';
+import {
+  RoutingService,
+  CustomerCouponService,
+  GlobalMessageService,
+  GlobalMessageType,
+  useFeatureStyles,
+} from '@spartacus/core';
+import { FocusConfig, LaunchDialogService } from '../../../../layout/index';
+import { ICON_TYPE } from '../../../../cms-components/misc/icon/index';
+
+@Component({
+  selector: 'cx-claim-dialog',
+  templateUrl: './claim-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ClaimDialogComponent implements OnDestroy, OnInit {
+  private subscription = new Subscription();
+  iconTypes = ICON_TYPE;
+  private pageSize = 10;
```

Comment:
`major:` let's make this field `protected` for easier extensibility and customisation

Reply:
thanks, fixed.